ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -919,25 +934,27 @@ const DataSetPage = fp.flow(withUserProfile(), withCurrentWorkspace(), withUrlPa
                       </div>
                     </div>
                   </BoxHeader>
-                  <div style={{height: valueSets.length > 0 ? '7.625rem' : '9rem', overflowY: 'auto'}}>
-                    {valuesLoading && <Spinner style={{position: 'relative',
+                  <div style={{height: selectedDomains.size > 0 ? '7.625rem' : '9rem', overflowY: 'auto'}}>
+                    {domainValueSetIsLoading.size > 0 && <Spinner style={{position: 'relative',
                       top: '2rem', left: 'calc(50% - 36px)'}}/>}
-                    {valueSets.map(valueSet =>
-                      <div key={valueSet.domain}>
+                    {Array.from(selectedDomains).map(domain =>
+                      domainValueSetLookup.has(domain) &&
+                      <div key={domain}>
                         <Subheader style={{fontWeight: 'bold'}}>
-                          {valueSet.survey ? 'Survey' : formatDomain(valueSet.domain)}
```

Comment:
@NehaBroad @blrubenstein  I could not understand what was going on here with this `.survey` value. I didn't find any comments hinting at the rationale for this. I removed it in my PR because it was a potential source of bugs/inconsistency. If this is tracking a real requirement, let me know.

Relatedly, I'm confused by this line: https://github.com/all-of-us/workbench/blob/master/ui/src/app/pages/data/data-set/dataset-page.tsx#L1010

Reply:
It might be because Survey was not initially not a DOMAIN hence we were checking if survey object is populated treat it as SURVEY

Reply:
Could the OBSERVATION thing be because of api/src/main/java/org/pmiops/workbench/db/dao/ConceptSetDao.java L26?

.put(Domain.SURVEY, "observation")

Reply:
Surveys used to be in the Observation domain, I've removed the special handling of this for now, since presumably we could have a concept set in the observation domain as well, and we'd want it to be labeled accordingly.